clients/go/commands/createJob_command.go

```diff
@@ -0,0 +1,144 @@
+package commands
+
+import (
+    "context"
+    "fmt"
+    "github.com/zeebe-io/zeebe/clients/go/pb"
+    "github.com/zeebe-io/zeebe/clients/go/utils"
+    "time"
+)
+
+type DispatchCreateJobCommand interface {
+    Send() (*pb.CreateJobResponse, error)
+}
+
+type CreateJobCommandStep1 interface {
+    JobType(string) CreateJobCommandStep2
+}
+
+type CreateJobCommandStep2 interface {
+    DispatchCreateJobCommand
+
+    SetCustomHeaderFromString(string) (CreateJobCommandStep2, error)
+    SetCustomHeaderFromStringer(fmt.Stringer) (CreateJobCommandStep2, error)
+    SetCustomHeaderFromMap(map[string]interface{}) (CreateJobCommandStep2, error)
+    SetCustomHeaderFromObject(interface{}) (CreateJobCommandStep2, error)
+
+    AddCustomHeader(string, interface{}) CreateJobCommandStep2
+
+    Retries(int32) CreateJobCommandStep2
+
+    PayloadFromString(string) (CreateJobCommandStep2, error)
+    PayloadFromStringer(fmt.Stringer) (CreateJobCommandStep2, error)
+    PayloadFromMap(map[string]interface{}) (CreateJobCommandStep2, error)
+    PayloadFromObject(interface{}) (CreateJobCommandStep2, error)
+}
+
+type CreateJobCommand struct {
+    utils.SerializerMixin
+
+    customHeaders map[string]interface{}
+
+    request       *pb.CreateJobRequest
+    gateway       pb.GatewayClient
+
+}
+
+func (cmd *CreateJobCommand) GetRequest() *pb.CreateJobRequest {
```

Comment:
Do we need this method only for testing? If so can we not make it public API? If it is just for testing the custom headers, and we can't make it private API, then I would prefer to create an issue to remove it and test custom headers using job workers if we implemented it.

Reply:
Why is it a problem? It's not private cause tests are in another module so in order to use it in tests it has to be public. 

But it's hidden away from the user behind the interface.

I would argue that other requests should have this design and create a ticket to this there in case. You can implement request logging on the workers much easier if you have some kind of access to the constructed request.